### ISSUE TITLE
Segmentation fault on OS X 10.11 with Python 3.10

### ISSUE THREAD

**Issue by BLKSerene** [pydantic/pydantic#7170]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Hi, it seems that `pydantic` v2 does not work on old macOSes such as El Capitan 10.11 with Python 3.10.

I suspect that the problem is that `pydantic` v2 introduced `pydantic_core` which is writen in Rust, and the Rust version used to build `pydantic_core` does not target at old macOSes.

`pydantic` officially support Python 3.7+ which then officially support macOS 10.9+. If there are requirements on the minimum version of macOS, a note in README would be helpful.

A similar issue found in other projects: https://github.com/undertheseanlp/underthesea/issues/677

### Example Code

```Python
import pydantic # A simple import would fail, so nothing of v2 works on OS X 10.11
```

### Error message:
Segmentation fault: 11

### Python, Pydantic & OS Version

```Text
Python: 3.10.11
Pydantic: 2.2.0
OS: OS X El Capitan 10.11.6
```

Selected Assignee: @hramezani

**Comment by hramezani** [comment]
Thanks @BLKSerene for reporting this 🙏 

@samuelcolvin , @davidhewitt do you have any idea?

**Comment by davidhewitt** [comment]
@BLKSerene are you able to share any more details about the crash, maybe a detailed crash log?

**Comment by BLKSerene** [comment]
@davidhewitt The only error message shown is (when simply importing `pydantic` on OS X 10.11):

`Segmentation fault: 11`

Is there any way in which I could get a more detailed crash log?

**Comment by davidhewitt** [comment]
Can you find them here? https://www.maketecheasier.com/read-macos-crash-reports-troubleshoot-mac/

**Comment by davidhewitt** [comment]
Thanks. Reading that doesn't give me any substantial clues unfortunately. Given that it's fixed in 2.2.1 I suspect this was resolved by the removal of `mimalloc` (https://github.com/pydantic/pydantic-core/pull/900). Not much else was changed from 2.2.0.

I think realistically we would struggle to test this, because we use the Github runners and if we tried to test all arch/os/python combinations it would be extremely broad. I'm going to close this as it sounds like the fix has already landed.

(It's not expected that we will add `mimalloc` again, so please ping if this recurs.)
